File: crates/domain/src/playback/watched_rollup.rs
use crate::playback::WatchTarget;

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct WatchedRollup {
    pub target: WatchTarget,
    pub watched: bool,
    pub completed: bool,
    pub watched_episodes: u32,
    pub total_episodes: u32,
}

File: clients/flutter/lib/model/user_library/watched_rollup.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:shadowmask/model/common/title_ref.dart';

part 'watched_rollup.freezed.dart';
part 'watched_rollup.g.dart';

@freezed
abstract class WatchedRollup with _$WatchedRollup {
  const factory WatchedRollup({
    required TitleRef target,
    @Default(false) bool watched,
    @Default(false) bool completed,
    @Default(0) int watchedEpisodes,
    @Default(0) int totalEpisodes,
  }) = _WatchedRollup;

  factory WatchedRollup.fromJson(Map<String, dynamic> json) =>
      _$WatchedRollupFromJson(json);
}
